lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
```

Comment:
(Though then we start getting into "what happens when we have a race condition", since this version of the repl is still very susceptible to that)